### PR TITLE
[clang-tidy] Fix support for typedefs in readability-identifier-naming

### LINKED ISSUE #54699: parameter types are not renamed when fixing readability-identifier-naming.TypedefCase
Using clang-tidy, I try to rename typedefs, and find that clang-tidy fixes the header and variable declarations, but not parameter types in function signatures.
It is similar to this issue: https://github.com/llvm/llvm-project/issues/47147
I could not find other open issue, so I report it.

Using clang+llvm-12.0.1-x86_64-linux-gnu-ubuntu-, working with c .h and .c files.

### .clang-tidy
```
Checks: "-*,readability-identifier-naming"
CheckOptions:
    - key:      readability-identifier-naming.TypedefCase
      value:    'camelBack'
    - key:      readability-identifier-naming.TypedefSuffix
      value:    '_t'
```

### test.h
```
typedef struct {
    int x;
} ABC;

```
### test.c
```
#include "test.h"
void fun(ABC a) {
    ABC b;
    size s=sizeof(ABC);
    ABC *c  = (ABC *)NULL;
}
```

is rewritten
### test.h
```
typedef struct {
    int x;
} abc_t;

```
### test.c
```
#include "test.h"
void fun(ABC a) {
    abc_t b;
    size s=sizeof(ABC);
    ABC *c  = (ABC *)NULL;
}
```

Type of parameter `a` not renamed.
Type ABC in sizeof is not renamed.
Pointer Type ABC*  is not renamed.
Type ABC in cast is not renamed.

### LINKED ISSUE #55156: [clang-tidy] Fails to detect readability-identifier-naming.TypedefPrefix check in case of macro usage
Tested with 13.0.0 and 14.0.0 and does not detect invalid case style for typedef!

Command line: `clang-tidy --config="{Checks: '-*,readability-identifier-naming', CheckOptions: [{key: readability-identifier-naming.TypedefPrefix, value: t_}]}" clang-tidy_typedef_bug.c`

When macro `STATIC_MACRO` is defined as `static` no warning is issued. When macro `STATIC_MACRO` is empty the check finds the invalid case style!

```c
typedef enum {
    e_value_0,
    e_value_1,
    e_value_2,
    e_value_3,
    e_value_4,
    e_value_5
} e_values_type;

#define STATIC_MACRO static

STATIC_MACRO void some_func(e_values_type values)
{
    // do something
}
```

### PR DESCRIPTION
**Pull request by PiotrZSL**
Typedef rename were not properly handled when typedef were used behind pointer, or as a part of function type. Additionally because entire function were passed as an source-range, when function started with macro, such change were not marked for a fix.

Removed workaround and used proper TypedefTypeLoc instead.

Fixes #55156, #54699

### CODE REVIEW

File: clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp [hunk 1]
```diff
@@ -729,3 +729,21 @@ struct forward_declared_as_struct;
...
+  VALUE1,
+} ValueType;
+// CHECK-MESSAGES: :[[@LINE-1]]:3: warning: invalid case style for typedef 'ValueType' [readability-identifier-naming]
+// CHECK-FIXES: {{^}}} value_type_t;
+
+#define STATIC_MACRO static
+STATIC_MACRO void someFunc(ValueType a_v1, const ValueType& a_v2) {}
+// CHECK-FIXES: {{^}}STATIC_MACRO void someFunc(value_type_t a_v1, const value_type_t& a_v2) {}
+STATIC_MACRO void someFunc(const ValueType** p_a_v1, ValueType (*p_a_v2)()) {}
+// CHECK-FIXES: {{^}}STATIC_MACRO void someFunc(const value_type_t** p_a_v1, value_type_t (*p_a_v2)()) {}
```
Comment (carlosgalvezp) [inline clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp:745]: Shouldn't we add `CHECK-MESSAGES` here as well?

File: clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp [hunk 2]
```diff
@@ -729,3 +729,21 @@ struct forward_declared_as_struct;
 class forward_declared_as_struct {
 };
 
+namespace pr55156 {
+
+typedef enum {
+  VALUE0,
+  VALUE1,
+} ValueType;
```
Comment (HerrCai0907) [inline clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp:737]: Could you also add some test cases for function typedef?

**Review by HerrCai0907 — APPROVED** [review]
LGTM

File: clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp (hunk 1 above)
Comment (carlosgalvezp) [reply clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp:751]: Ack!

**Review by carlosgalvezp — APPROVED** [review]
LGTM, thanks for the fix and the great cleanup!

**Pull request merged**